snapcraft/plugins/kbuild.py

```diff
@@ -166,13 +169,19 @@ def do_install(self):
                  ['CONFIG_PREFIX={}'.format(self.installdir)] +
                  self.make_install_targets)
 
-    def build(self):
+    def prep_build(self):
         super().build()
 
-        config_path = os.path.join(self.builddir, '.config')
+        config_path = self.get_config_path()
 
         self.do_base_config(config_path)
         self.do_patch_config(config_path)
         self.do_remake_config()
+
+    def finish_build(self):
         self.do_build()
         self.do_install()
+
+    def build(self):
+        self.prep_build()
+        self.finish_build()
```

Comment:
prepare and finish sounds like a poor separation for this function. Maybe we can find better names. I would like something like:
```

def build(self):
    self.configure()
    self.do_build()
    self.do_install()
```

It's still poor naming because inside build you have another call to build, but that's because snapcraft doesn't yet separate the build and install phases. I hope we can improve that soon.